skimage/feature/tests/test_util.py

```diff
@@ -68,6 +68,9 @@ def test_mask_border_keypoints():
 )
 def test_plot_matches(shapes):
     from matplotlib import pyplot as plt
+    from matplotlib import use
+
+    use('Agg')
```

Comment:
Even with the `MPLBACKEND` environment variable set inside the workflow, it didn't work and the tests for Matplotlib were failing – I would suggest keeping this change here (and in other places).

Reply:
I won't have as much context as @lagru here, but for what it's worth: in SciPy we set the backend to Agg for testing, and it's my understanding that that in general is a robust strategy.

Reply:
I don't really follow why `use('Agg')` fixes things and `MPLBACKEND="Agg"` doesn't. E.g. for https://github.com/scikit-image/scikit-image/pull/7350/commits/be525a44f4620ee423dbd47e521c87b0bd4c3fd8, [pytest's error log](https://github.com/scikit-image/scikit-image/actions/runs/8360772772/job/22887205529#step:9:1875) already indicates that `Agg` is used.

This is before you try to [use Agg programmatically](https://github.com/scikit-image/scikit-image/pull/7350/commits/f466193dea59ce92d50ce643f9a0c70a4f7b9386)

Reply:
I don't understand it enough either (perhaps Node.js is not properly letting environment variables pass to the Python interpreter). But sure, let me try to revert this change in a separate commit, again, at the end of addressing the review and we can see if things break in an isolated state. Either way, keeping just one of these options would be better.

Reply:
I have done a bit of reading about this as someone who does not have much JavaScript experience, and I think what is happening here is that Pyodide runs Python code within the browser environment, isolated from the Node.js process that spawns the Pyodide interpreter – therefore, it doesn't directly inherit environment variables from Node.js.
It could be tricky to debug further but I'm happy to do so, so I'll remove the `MPLBACKEND` environment variable from the workflow file for now, given that the programmatic approach with `matplotlib.use` will be beneficial for not only these Pyodide-specific tests, but for the general test suite on regular supported platforms too.